packages/application-shell/src/components/login-sso-callback/login-sso-callback.js

```diff
@@ -31,16 +28,20 @@ export class LoginSSOCallback extends React.PureComponent {
     const fragments = qs.parse(this.props.location.hash.substring(1));
     const idToken = fragments.id_token;
     const decodedIdToken = jwtDecode(idToken);
-    const nonce = storage.get(STORAGE_KEYS.NONCE);
-    storage.remove(STORAGE_KEYS.NONCE);
+    const sessionState = storage.get(
+      `${STORAGE_KEYS.NONCE}_${decodedIdToken.nonce}`,
+      { storage: window.sessionStorage }
+    );
+    // Clear the nonce, we don't need it anymore
+    storage.remove(STORAGE_KEYS.NONCE, { storage: window.sessionStorage });
```

Comment:
I always felt wrapping a native API only makes sense if:

1. Eases testing eg mocks to be used
2. It is needed and easy to swap the storage wihout having to change the call-site

Neither holds for us. Jest has great mocking plugins for local storage etc. Plus we do not really want to or need to swap the engine (as in strategy pattern) here. 

Anyway I am fine with keeping thigs obviously. However we can ask how much we need the package.

Reply:
Yeah. Are you referring to this package? https://www.npmjs.com/package/jest-localstorage-mock?

Reply:
Yes and I think it could be easily extended or duped for session storage. 

Which is why I‘d wish us go use

1. Browser APIs directly instead of mapping and renaming or
2 Build on our existing core storage package instead of adding another abstraction here which may just be confusing as to which one exists for what reason